### PR TITLE
SONARJAVA-988 Support detection in nested ifs

### PR DESCRIPTION
**Pull request by merciesa**

### CODE REVIEW

File: java-checks/src/test/files/checks/UselessConditionCheck.java [hunk 1]
```diff
@@ -27,6 +27,13 @@ public void method(Object foo, Object bar, boolean condition) {
     }
     if (!(foo == bar || condition) || (foo == bar || condition)) { // Noncompliant, always true
     }
+    if (foo == bar) {
+      if (foo == bar) { // Noncompliant, always true
```
Comment (benzonico) [inline java-checks/src/test/files/checks/UselessConditionCheck.java]: I am pretty sure your detector is way smarter than this  ! :)

File: java-checks/src/main/java/org/sonar/java/checks/UselessConditionCheck.java [hunk 2]
```diff
@@ -53,18 +61,43 @@ public void scanFile(JavaFileScannerContext context) {
...
+  private void checkIf(List<ExecutionState> states, IfStatementTree tree) {
+    ExpressionEvaluatorVisitor results = checkCondition(states, tree.condition());
+    checkNestedIf(results.trueStates, tree.thenStatement());
+    if (tree.elseStatement() != null) {
+      checkNestedIf(results.falseStates, tree.elseStatement());
+    }
+  }
+
+  private void checkNestedIf(List<ExecutionState> states, Tree tree) {
+    if (tree.is(Tree.Kind.BLOCK)) {
```
Comment (benzonico) [inline java-checks/src/main/java/org/sonar/java/checks/UselessConditionCheck.java]: Please do not reimplement visit of tree. 

This implementation fails to visit an if statement with an anonymous class implementation as then statement.

File: java-checks/src/main/java/org/sonar/java/checks/UselessConditionCheck.java [hunk 3]
```diff
@@ -53,18 +61,43 @@ public void scanFile(JavaFileScannerContext context) {
...
-  private void checkCondition(ExpressionTree tree) {
-    ExpressionEvaluatorVisitor evaluation = new ExpressionEvaluatorVisitor(new ExecutionState(), tree);
-    if(evaluation.isAlwaysFalse()) {
+  private ExpressionEvaluatorVisitor checkCondition(List<ExecutionState> states, ExpressionTree tree) {
+    ExpressionEvaluatorVisitor evaluation = new ExpressionEvaluatorVisitor(states, tree);
+    if (evaluation.isAlwaysFalse()) {
       raiseIssue(tree, "false");
     }
-    if(evaluation.isAwlaysTrue()) {
+    if (evaluation.isAwlaysTrue()) {
```
Comment (benzonico) [inline java-checks/src/main/java/org/sonar/java/checks/UselessConditionCheck.java]: typo in method name.

**Comment by merciesa** [comment]
Functionality covered in SONARJAVA-989

**Pull request closed without merging**